tests/test_dispatcher/test_middlewares/test_user_context.py

```diff
@@ -41,3 +42,46 @@ async def test_call(self):
         assert data["event_chat"] is chat
         assert data["event_from_user"] is user
         assert data["event_thread_id"] == thread_id
+
+    async def test_resolve_event_context(self):
+        middleware = UserContextMiddleware()
+        data = {}
+
+        chat = Chat(id=1, type="private", title="Test")
+        user = User(id=2, first_name="Test", is_bot=False)
+        add_date = datetime.now()
+        expiration_date = datetime.now()
+
+        source = ChatBoostSourcePremium(user=user)
+        boost = ChatBoost(boost_id="Test", add_date=add_date, expiration_date=expiration_date, source=source)
+        update = Update(update_id=42, chat_boost=ChatBoostUpdated(chat=chat, boost=boost))
+
+        await middleware(next_handler, update, data)
+
+        event_context = data["event_context"]
+        assert isinstance(event_context, EventContext)
+        assert event_context.chat is chat
+        assert event_context.user is user
+        assert event_context.thread_id is None
+        assert data["event_chat"] is chat
+        assert data["event_from_user"] is user
+        assert data.get("event_thread_id") is None
+
+        data.clear()
+
+        source = ChatBoostSourceGiftCode(user=user)
+        boost = ChatBoost(boost_id="Test", add_date=add_date, expiration_date=expiration_date, source=source)
+        update = Update(update_id=42, chat_boost=ChatBoostUpdated(chat=chat, boost=boost))
+
+        await middleware(next_handler, update, data)
+
+        event_context = data["event_context"]
+        assert isinstance(event_context, EventContext)
+        assert event_context.chat is chat
+        assert event_context.user is None
+        assert event_context.thread_id is None
+        assert data["event_chat"] is chat
+        assert data.get("event_from_user") is None
+        assert data.get("event_thread_id") is None
+
+        data.clear()
```

Comment:
Use parametrize to repeat the test with other params

Reply:
Also try to find other tests covering neighbour data in context. Looks like we just need to add params to existing test

Reply:
I don't understand what you mean by 
> Looks like we just need to add params to existing test

`test_call` test or the test that I wrote?
I don't see the point in the first test and how to cover the current case.